docs/architecture/data-plane-osd-cluster-dynamic-scaling.md

```diff
@@ -6,18 +6,181 @@ This documents explains how Data Plane OSD Cluster Dynamic Scaling functionality
 has been implemented.
 
 Data Plane OSD Cluster Dynamic Scaling functionality currently deals with:
-* Autoscaling of compute nodes of an OSD cluster
-* Prewarming of compute nodes 
+* Autoscaling of nodes of an OSD cluster
+* Prewarming of nodes 
 * OSD Cluster creation and deletion
 
-## Autoscaling of compute nodes of an OSD cluster
+## Autoscaling of worker nodes of an OSD cluster
 
-<!---TODO -->
+Autoscaling of worker nodes of an OSD cluster is done by leveraging the [Cluster Autoscaler](https:docs.openshift.com/container-platform/4.9/machine_management/applying-autoscaling.html) as described in https:architecture.appservices.tech/ap/15/#autoscaling-of-nodes. For Manual clusters, this has to be enabled manually. Worker node autoscaling is enabled by default for all clusters that are created dynamically by the Fleet manager
 
-## Prewarming of compute nodes 
+## Prewarming of worker nodes 
 
-<!---TODO -->
+Data Plane Worker Node Prewarming helps speed up the allocation of Kafka instances in the Data Planes managed by the control plane in some scenarios. It
+is achieved by reserving resources in the Data Planes. The reservation of resources is specified in the form of reserved Kafka instances of a given
+size and it is done on a per Kafka instance type basis.
+
+When enabling and configuring Data Plane Node Prewarming for a given Kafka instance type, each Data Plane managed by the Control Plane maintains the
+specified reserved capacity for that instance type at all times. In this way, if the available resources in the existing worker nodes of a Data Plane are not
+enough to contain the reserved capacity then the number of worker nodes is automatically scaled up to be able to maintain the reserved capacity. By doing
+so, the worker nodes can potentially be automatically scaled up ahead of time. In that case, when a Kafka instance is created there is no need to scale up
+additional worker nodes and wait for them to be created.
+
+>NOTE: Worker node horizontal scale up within a Data Plane cluster is limited to the maximum number of supported worker nodes supported by an OSD cluster in OCM.
+
+Data Plane Worker Node Prewarming is used to mitigate the wait time of Kafka instances provisioning due to new worker nodes being created for a given data
+plane. How effectively it mitigates that wait time depends on the amount of reserved capacity. The more reserved capacity, the more effective
+Data Plane Worker Node prewarming will be against actions like bursts of Kafka instance creations.
+
+Thus, the decision on how much reserved capacity is configured is a tradeoff between:
+
+* Cost: The more reserved capacity the more resources are consumed and thus have to be paid for
+* The desired likelihood of new Kafka Instances creation having to wait for new worker nodes to be available so they can be allocated
+
+### Configuration
+
+Fleet manager exposes the following configuration.
```

Comment:
This is mostly talking about how to configure it via the template. Do you think it's worth calling out the [config file](https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/main/config/node-prewarming-configuration.yaml) when running locally?

Reply:
+1 

I’ll add it

Reply:
I've updated it.